### ISSUE TITLE
Add virtual field pattern

### ISSUE THREAD

**Issue by SupaHam** [iluwatar/java-design-patterns#52]
There's a new (I think) pattern called the Virtual field pattern, introduced in Java 8. https://kerflyn.wordpress.com/2012/07/09/java-8-now-you-have-mixins/ Scroll half way through till you find "Another Approach".

**Comment by iluwatar** [comment]
Thanks, we'll look into it.

**Comment by iluwatar** [comment]
Ok I've read the article and can see what can be done with virtual extension methods. 

There is discussion about it on [this mailing list](http://mail.openjdk.java.net/pipermail/lambda-dev/2012-July/005165.html) and it seems that Mr. Brian Goetz, who is the Oracle's Java Language Architect, is strongly against using Java 8 this way, for behavior and state multiple inheritance.

So I'm thinking that this is hardly a best practise or recommended solution a problem.

**Comment by iluwatar** [comment]
Reopened, see #638

**Comment by reegnz** [comment]
I will implement an example for this. But first I need to go read your contribution guidelines.

**Comment by iluwatar** [comment]
Ok @reegnz

**Comment by iluwatar** [comment]
@reegnz are you still implementing this?

**Comment by reegnz** [comment]
Slowly but surely.

**Comment by iluwatar** [comment]
@reegnz how slowly? I will mark this issue ready for taking again.

**Comment by reegnz** [comment]
No problem. I don't develop Java nowadays.

**Comment by ramazansakin** [comment]
Hi @iluwatar ,

I can add a simple app for virtual-pattern but when I studied the code a little, I saw some files that need to be added. How can I create these files. Or if I just add the sample project, would it be enough?

**Comment by iluwatar** [comment]
What files do you mean? I expect that implementing this pattern brings quite many new files anyway. You're not talking about source code files, are you?

**Comment by ramazansakin** [comment]
I made a png file for uml diagrams but not similar with other samples and .puml file and also I saw a block almost every sample as "The MIT License
 * Copyright © 2014-2019 Ilkka Seppälä".
Is this necessary to add any sample ?

**Comment by iluwatar** [comment]
puml file is autogenerated from code. From that you can generate the image and it will be similar as in other pattern examples.

Similarly there is a license checker plugin in the project that will add the missing licenses for you.

**Comment by ramazansakin** [comment]
@iluwatar  Yes, I saw the other samples, and I researched a little and I found plantUml plugin and I tried to generate them but still I can not auto generate them. Also, I researched about auto generating for uml and png part but still could not. I implement a sample for virtual field pattern and also blackboard but just left etc folders

**Comment by iluwatar** [comment]
Can you give a link to your code? I could take a look.

**Comment by ramazansakin** [comment]
@iluwatar , I forked the project and added a module as virtual-field, and I can not push my changes its getting denied error. Some permission problems or anything else ?

**Comment by iluwatar** [comment]
There shouldn't be any problems if you are trying to push to your forked repository. I can imagine that it fails if you are trying to push to the master repository - that's only allowed for collaborators.

**Comment by iluwatar** [comment]
More information about the pattern: https://stackoverflow.com/questions/47101072/java-virtual-field-pattern
